tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrdersControllerTests.cs

```diff
@@ -31,130 +30,107 @@ public void Constructor_NullRepository_Throws()
         [Test]
         public async Task GetAllAsync_NoOrdersExist_ReturnsEmptyResult()
         {
-            using var controller = OrdersControllerBuilder
-                .Create()
-                .WithListOrders(new List<Order>())
-                .Build();
-
-            var result = await controller.GetAllAsync();
+            var context = OrdersControllerTestContext.Setup();
 
-            result.Should().BeOfType<OkObjectResult>();
-            result.Should().NotBeNull();
+            using var controller = context.OrdersController;
 
-            var objectResult = result as OkObjectResult;
-            objectResult.Value.Should().BeOfType<List<OrderModel>>();
-
-            var ordersResult = objectResult.Value as List<OrderModel>;
-            ordersResult.Count().Should().Be(0);
+            var result = await controller.GetAllAsync(Guid.Empty) as OkObjectResult;
+            var orders = result.Value as List<OrderModel>;
+            orders.Count.Should().Be(0);
         }
 
         [TestCase(null, null)]
         [TestCase("C0000014-01", "Some Description")]
         [TestCase("C0000014-02", null)]
-        public async Task GetAllAsync_SingleOrderExists_ReturnsTheOrder(string orderId, string orderDescription)
+        public async Task GetAllAsync_SingleOrderWithOrganisationIdExists_ReturnsTheOrder(string orderId, string orderDescription)
         {
-            var model = new List<Order>()
-            {
-                new Order()
-                {
-                    OrderId = orderId,
-                    Description = orderDescription,
-                    LastUpdatedBy = Guid.NewGuid(),
-                    LastUpdated = DateTime.UtcNow,
-                    Created = DateTime.UtcNow,
-                    OrderStatus = submittedStatus
-                }
-            };
+            var organisationId = Guid.NewGuid();
 
-            var expected = new List<OrderModel>()
+            var orders = new List<(Order order, OrderModel expected)>
             {
-                new OrderModel()
-                {
-                    OrderId = model[0].OrderId,
-                    OrderDescription = model[0].Description,
-                    LastUpdatedBy = model[0].LastUpdatedBy,
-                    LastUpdated = model[0].LastUpdated,
-                    DateCreated = model[0].Created,
-                    Status = model[0].OrderStatus.Name
-                }
+                CreateOrderTestData(orderId, organisationId, orderDescription)
             };
 
+            var context = OrdersControllerTestContext.Setup();
+            context.Orders = orders.Select(x => x.order);
+
+            using var controller = context.OrdersController;
+
+            var result = await controller.GetAllAsync(organisationId) as OkObjectResult;
+            var ordersResult = result.Value as List<OrderModel>;
+            ordersResult.Count.Should().Be(1);

```

Comment:
Again, I'd suggest using FluentAssertions' method
`ordersResult.Count.Should().Be(1);` -> `ordersResult.Should().ContainSingle();`

Reply:
👍